hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServicePrecompileTest.java

```diff
@@ -294,269 +1328,161 @@ void createFungibleTokenWithInheritKeysEstimateGas() {
+        final var contract = testWeb3jService.deploy(ModificationPrecompileTestContract::deploy);
+        final var function = contract.send_createFungibleTokenExternal(
+                convertTokenEntityToHederaToken(domainBuilder
+                        .token()
+                        .customize(t -> t.initialSupply(10_000_001L))
+                        .get()),
+                INITIAL_SUPPLY,
+                DECIMALS,
+                VALUE);
+
+        assertThatThrownBy(function::send).isInstanceOf(MirrorEvmTransactionException.class);
+    }
+
+    @Test
+    void createNftWithNoSupplyKeyFails() {
+        final var contract = testWeb3jService.deploy(ModificationPrecompileTestContract::deploy);
+        final var function = contract.send_createNonFungibleTokenExternal(
+                convertTokenEntityToHederaToken(domainBuilder
+                        .token()
+                        .customize(t -> t.supplyKey(new byte[0]))
+                        .get()),
+                VALUE);
+
+        assertThatThrownBy(function::send).isInstanceOf(MirrorEvmTransactionException.class);
+    }
+
+    @Test
+    void createNftWithNoTreasuryFails() {
+        final var contract = testWeb3jService.deploy(ModificationPrecompileTestContract::deploy);
+        final var function = contract.send_createNonFungibleTokenExternal(
+                convertTokenEntityToHederaToken(domainBuilder
+                        .token()
+                        .customize(t -> t.treasuryAccountId(null))
+                        .get()),
+                VALUE);
+
+        assertThatThrownBy(function::send).isInstanceOf(MirrorEvmTransactionException.class);
+    }
+
+    @Test
+    void createNftWithDefaultFreezeWithNoKeyFails() {
+        final var contract = testWeb3jService.deploy(ModificationPrecompileTestContract::deploy);
+        final var function = contract.send_createNonFungibleTokenExternal(
+                convertTokenEntityToHederaToken(domainBuilder
+                        .token()
+                        .customize(t -> t.freezeDefault(true).freezeKey(new byte[0]))
+                        .get()),
+                VALUE);
+
+        assertThatThrownBy(function::send).isInstanceOf(MirrorEvmTransactionException.class);
+    }
+
+    @Test
+    void createNftWithInvalidAutoRenewAccountFails() {
+        final var contract = testWeb3jService.deploy(ModificationPrecompileTestContract::deploy);
+        final var function = contract.send_createNonFungibleTokenExternal(
+                convertTokenEntityToHederaToken(
+                        domainBuilder.token().get(),
+                        domainBuilder
+                                .entity()
+                                .customize(e -> e.autoRenewPeriod(10L))
+                                .get()),
+                VALUE);
+
+        assertThatThrownBy(function::send).isInstanceOf(MirrorEvmTransactionException.class);
```

Comment:
For refactored tests in this class, we should follow the given/when/then syntax. It will help with readability especially here since we're coupling the given and the when too tightly.

```suggestion
        // Given
        final var contract = testWeb3jService.deploy(ModificationPrecompileTestContract::deploy);
        var entity = domainBuilder.entity().customize(e -> e.autoRenewPeriod(10L)).get();
        var token = domainBuilder.token().get();
        var hederaToken = convertTokenEntityToHederaToken(token, entity);

        // When
       var function = contract.send_createNonFungibleTokenExternal(hederaToken, BigInteger.valueOf(10));

        // Then
        assertThatThrownBy(function::send).isInstanceOf(MirrorEvmTransactionException.class);
```